### DISCUSSION TITLE
Some question about session_token

### DISCUSSION THREAD

**Discussion by Harry-Jing** [acheong08/ChatGPT#202] [Q&A] [answered]
I want to know about how long the token will expire if I keep refreshing it

**Comment by acheong08** [comment, marked as answer]
It is currently unknown. The cookie expiration says one month but the API might handle things differently

**Comment by Harry-Jing** [comment]
If each refresh extends the expiration time, is it possible to give the function that refreshes the token a value that returns the new token.

**Reply by acheong08** [reply]
Each refresh doesn't seem to increase expiration time. I think the whole thread of tokens is set to expire in one month at which point you need to authenticate via email/password.

**Reply by Harry-Jing** [reply]
My token feels like it can easily expire, usually only 1 day. Because I create a new Chatbot object every time. So I want to save the token for each refresh to config

**Reply by acheong08** [reply]
Tokens expire if not refreshed

**Reply by Harry-Jing** [reply]
I think this is exactly where my problem is

**Reply by acheong08** [reply]
If email/password also defined, it should refresh automatically if it detects that they have expired

**Reply by Harry-Jing** [reply]
problem is my country is not supported area by openai

**Reply by Harry-Jing** [reply]
and if I use proxy, the captcha will be require

**Reply by acheong08** [reply]
Same. I just use VPN to bypass. (You're in China as well right?)

**Reply by acheong08** [reply]
> and if I use proxy, the captcha will be require

Check out https://2captcha.com/lang/python which you can implement into the captcha solving system

**Reply by acheong08** [reply]
It becomes automatic if you use it

**Reply by acheong08** [reply]
Cost: https://2captcha.com/pricing -- Normal captcha

**Comment by Harry-Jing** [comment]
Can it now automatically pass captcha？

**Reply by Harry-Jing** [reply]
I haven't test yet

**Reply by acheong08** [reply]
The CLI saves the captcha to a file which you then have to enter into the terminal

**Reply by acheong08** [reply]
It requires you to write your own custom captcha handler as a developer (which you can choose what API you use)

**Reply by acheong08** [reply]
No docs yet. You can see the commit here: https://github.com/acheong08/OpenAIAuth/commit/24886c94597d6535402c7eba9d3bef95de95ba76

**Reply by acheong08** [reply]
https://github.com/acheong08/ChatGPT/blob/a054c4e0f112ded99a7df511c4af5c87b2517760/src/revChatGPT/__main__.py#L137-L161

This is how I implemented it as an example in main

**Reply by acheong08** [reply]
And used in Chatbot class https://github.com/acheong08/ChatGPT/blob/a054c4e0f112ded99a7df511c4af5c87b2517760/src/revChatGPT/__main__.py#L55-L56

**Reply by Harry-Jing** [reply]
I got it
